wal/src/tests/read_write.rs

```diff
@@ -15,6 +15,130 @@ use crate::{
     },
 };
 
+#[test]
+fn test_rocksdb_wal() {
+    let builder = RocksWalBuilder::default();
+
+    test_all(builder);
+}
+
+#[test]
+fn test_memory_table_wal_default() {
+    let builder = MemoryTableWalBuilder::default();
+
+    test_all(builder);
+}
+
+#[test]
+fn test_memory_table_wal_with_ttl() {
+    let builder = MemoryTableWalBuilder::with_ttl("1d");
+
+    test_all(builder);
+}
+
+fn test_all<B: WalBuilder>(builder: B) {
```

Comment:
I guess `test_all` is not a best practice for unit test because it is hard to find the problems in such a long procedure when some failure happens.